addons/website_sale/controllers/main.py

```diff
@@ -535,11 +535,33 @@ def checkout_values(self, **kw):
         return values
 
     def _get_mandatory_billing_fields(self):
+        # deprecated for _get_mandatory_fields_billing which handle zip/state required
         return ["name", "email", "street", "city", "country_id"]
 
     def _get_mandatory_shipping_fields(self):
+        # deprecated for _get_mandatory_fields_shipping which handle zip/state required
         return ["name", "street", "city", "country_id"]
 
+    def _get_mandatory_fields_billing(self, country_id=False):
```

Comment:
`_get_mandatory_fields_billing` --> `_get_mandatory_all_shipping_fields`

Reply:
:+1:

Reply:
@JKE-be it is a little ugly to have `_get_mandatory_billing_fields` and `_get_mandatory_fields_billing`. (not easy to understand the difference ...)

Reply:
You will remove `_get_mandatory_billing_fields` in master ?

Reply:
as you say (and i is commented), it is purely for retrocompatibility. During fwdport to master it will be just dropped

Reply:
yes, it is the idea